src/containers/App/component.js

```diff
@@ -64,12 +46,13 @@ export default class App extends React.Component {
   }
 
   getChildContext () {
-    const { dispatch, currentUser, isMobile, location } = this.props
-    return {dispatch, currentUser, isMobile, location}
+    const { currentUser, isMobile, location } = this.props
+    const { dispatch } = this.context
```

Comment:
App doesn't have dispatch in its context -- it's the component that puts it into context for everyone else -- you have to get it from props.